### PR TITLE
Run get_signals async.

### PR DESCRIPTION
**Pull request by pan-long**
This should speed up the check and analyze signals phase.

### CODE REVIEW

**Comment by glonlas** [comment]
Thanks for this PR, I can’t wait to test it.
May we have a unitest that cover the exception that can be raised?

File: freqtrade/main.py [hunk 1]
```diff
@@ -209,8 +211,18 @@ def create_trade(stake_amount: float) -> bool:
         raise DependencyException('No pair in whitelist')
 
     # Pick pair based on StochRSI buy signals
-    for _pair in whitelist:
-        if get_signal(_pair, SignalType.BUY):
+    with ThreadPoolExecutor() as pool:
+        awaitableSignals = []
```
Comment (gcarq) [inline freqtrade/main.py]: please use snake_case

File: freqtrade/analyze.py [hunk 2]
```diff
@@ -144,6 +144,9 @@ def get_signal(pair: str, signal: SignalType) -> bool:
     except ValueError as ex:
         logger.warning('Unable to analyze ticker for pair %s: %s', pair, str(ex))
         return False
+    except Exception as inst:
+        logger.warning('Unexpected error when analyzing ticker for pair %s: %s', pair, str(inst))
```
Comment (gcarq) [inline freqtrade/analyze.py]: I think we should log this broad exception with `logger.exception()` so that we have a stacktrace

**Comment by glonlas** [comment]
Hi Pan
Did you get any API invalid key message when using this PR overnight?

After 10h running this PR I had this error message:


Status: Got OperationalException:
Traceback (most recent call last):
  File "./freqtrade/main.py", line 83, in _process
    state_changed = create_trade(float(_CONF['stake_amount']))
  File "./freqtrade/main.py", line 222, in create_trade
    if exchange.get_balance(_CONF['stake_currency']) < stake_amount:
  File "freqtrade/exchange/__init__.py", line 129, in get_balance
    return _API.get_balance(currency)
  File "freqtrade/exchange/bittrex.py", line 84, in get_balance
    currency=currency))
freqtrade.OperationalException: APIKEY_INVALID params=(BTC)
Issue /start if you think it is safe to restart.

I don’t know if this is related to a threshold I reached, or a random Bittrex API error.

File: freqtrade/main.py [hunk 3]
```diff
@@ -209,8 +211,18 @@ def create_trade(stake_amount: float) -> bool:
...
+        for _pair in whitelist:
```
Comment (gcarq) [inline freqtrade/main.py]: I would use a list comprehension to build the awaitable signals list, it's more pythonic, tends to perform better (depends on python implementation) and gets rid of intermediate variables

```
awaitable_signals = [
    asyncio.wrap_future(pool.submit(get_signal, pair, SignalType.BUY))
    for pair in whitelist
]
```

**Comment by pan-long** [comment]
Hi @glonlas, I didn't hit that error. I have been running my bot with this change for 3 days. It looks like some bittrex API error.

I will make the changes and update the PR.

File: freqtrade/analyze.py (hunk 2 above)
Comment (pan-long) [reply freqtrade/analyze.py]: done.

File: freqtrade/main.py (hunk 1 above)
Comment (pan-long) [reply freqtrade/main.py]: done.

File: freqtrade/main.py (hunk 3 above)
Comment (pan-long) [reply freqtrade/main.py]: done.

**Comment by pan-long** [comment]
Hi @gcarq, I have made the change and squashed the commits. You reviews are gone. Can you take a look of the commit again? Thanks!

**Pull request merged**